Core/NoOpRunner.Core/Shapes/Platform.cs

```diff
@@ -1,61 +1,98 @@
 ﻿using NoOpRunner.Core.Entities;
 using NoOpRunner.Core.Enums;
 using System;
+using System.Collections.Generic;
 
 namespace NoOpRunner.Core.Shapes
 {
     public class Platform : BaseShape
     {
-        private Random random = new Random();
         public int bottomPosY { get; set; } // Bottom of x platform
         public int topPosY { get; set; } // Max height of x platform
 
+        private Random randomNumber;
+
         public int lastPosX = 0,
                    lastPosY = 0;
 
-        public Platform(int centerPosX, int centerPosY, int bottomPosY, int topPosY) : base(centerPosX, centerPosY)
+        private List<int> xCoords = new List<int>();
+        private List<int> yCoords = new List<int>();
+
+        public Platform(int centerPosX, int centerPosY, int bottomPosY, int topPosY, Random random) : base(centerPosX, centerPosY)
         {
+            randomNumber = random;
+
             while (lastPosX < 28) // TODO
             {
                 int blockLength = randomLength(4);
                 int blockHeight = randomHeight(4); // Lower than jump height
 
-                MapShapeX(lastPosX, lastPosY, blockLength, Color.Red);
+                addShape(true, lastPosX, lastPosY, blockLength);
                 lastPosX += blockLength;
+
                 if(blockHeight < 0)
                 {
-                    MapShapeY(lastPosX, 1, lastPosY, Color.Red);
+                    addShape(false, lastPosX, 1, lastPosY);
                     lastPosY = 0;
                 } else
                 {
                     if (lastPosY + blockHeight > topPosY)
                     {
-                        MapShapeY(lastPosX, lastPosY, (topPosY - lastPosY), Color.Red);
+                        addShape(false, lastPosX, lastPosY, (topPosY - lastPosY) < bottomPosY ? bottomPosY : (topPosY - lastPosY));
                         lastPosY = topPosY;
                     } else
                     {
-                        MapShapeY(lastPosX, lastPosY, blockHeight, Color.Red);
+                        addShape(false, lastPosX, lastPosY, blockHeight);
                         lastPosY = lastPosY + blockHeight;
                     }
                 }
             }
         }
 
+        public void addShape(bool horizontal, int posX, int posY, int blockLength)
+        {
+            if (horizontal) MapShapeX(posX, posY, blockLength, Color.Red);
```

Comment:
Should put the statement after if clause into a new line for better readability, same for else